common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -317,6 +348,22 @@ void release(DefaultHandle<T> handle) {
             }
         }
 
+        void unguardedRelease(DefaultHandle<T> handle) {
+            handle.unguardedToAvailable();
+            Thread owner = this.owner;
+            if (owner != null && Thread.currentThread() == owner && batch.size() < chunkSize) {
```

Comment:
I think we can remove the null check here as `Thread.currentThread()` will never return null
```suggestion
            if (Thread.currentThread() == owner && batch.size() < chunkSize) {
```

Reply:
Or was this to eliminate some overhead when `owner` is null ?

Reply:
:) yep

Reply:
maybe adding a comment would make this more clear ;) ?

Reply:
This is a fair point sir 🤣

Reply:
I didn't modified this, but copied from @chrisvest original code, I'm now unifying both but I'll wait @chrisvest comment in case we can replace the `owner` null check with your suggestion

Reply:
I have no preference either way. Null checks are cheap. For platform threads at least, getting the current thread is also very fast.